### PR TITLE
Refs #29039 - Use foreman's Apache config standalone

### PR DESCRIPTION
**Pull request by ekohl**
By using only the Apache config, it's possible to use the Apache config without Foreman installed.

Depends on https://github.com/theforeman/puppet-foreman/pull/800

### CODE REVIEW

**Comment by ehelms** [comment]
The test failure look legitimate.

**Comment by ekohl** [comment]
That looks like it ran before the PR was merged so I've restarted the tests.

**Comment by ehelms** [comment]
Nah, its the same and valid failure. We dropped including `foreman` and there are references in `pulp.pp` to `foreman::user`.

**Comment by ekohl** [comment]
And that's why I hate two services sharing a filesystem ;)

Just thinking out loud: we have this because Katello export uses this.

https://github.com/Katello/katello/blob/acaacd8bf7ac463ab8a35bf37ec555d185cc3754/app/models/setting/content.rb#L94-L95

This is probably in `/var/lib/pulp` because it copies files and that's faster if it's on the same filesystem.

https://github.com/Katello/katello/blob/53f50325a502997ced2239f2ecf11d1d31d45121/app/lib/actions/katello/repository/export.rb#L58-L59

How big would the impact be if we moved this somewhere where Foreman can write and to the file `application.pp`?

Perhaps a cleaner way is to move it to `application.pp` regardless and implement an `anchor` which can be collected with an `exec` that does `mkdir -p` so the parent directory actually exists.

**Comment by ehelms** [comment]
Couple things to consider. First, Pulp does all the exporting so it has to be a location that Pulp (via the Apache user, although perhaps that should be the pulp user on newer installs) can write to. Hence the setup and location you see. We, as in Katello, when performing this sort of export then copy that Pulp export to a location of the users choosing. This is called the triple-disk-space problem with "legacy" Pulp exports.

Pulp 3 drops this export workflow in favor of a newer, core supported workflow.

My vote is to do the least amount of work to keep this workflow working as is since we will yank it all out after the Katello 3.16 release.

**Comment by ekohl** [comment]
Let's see if this works.

File: manifests/init.pp [hunk 1]
```diff
@@ -201,11 +208,4 @@
     repo_export_dir          => $repo_export_dir,
```
Comment (ehelms) [inline manifests/init.pp]: I assume this needs to be dropped

**Comment by ekohl** [comment]
An alternative is to drop this entire PR. Since the devel workflow doesn't use this class, it should have no effect there. If nightlies pass without this patch, we can drop this. Then after 3.16, it will be much simpler and the need is no longer there.

**Pull request merged**